packages/main/src/ComboBox.ts

```diff
@@ -61,277 +68,34 @@ import ComboBoxItem from "./ComboBoxItem.js";
-		 * @private
-		 */
-		_isValueStateFocused: {
-			type: Boolean,
-		},
-
-		/**
-		 * Defines the accessible ARIA name of the component.
-		 *
-		 * @type {string}
-		 * @defaultvalue: ""
-		 * @public
-		 * @since 1.0.0-rc.15
-		 */
-		accessibleName: {
-			type: String,
-			defaultValue: undefined,
-		},
-
-		/**
-		 * Receives id(or many ids) of the elements that label the component
-		 * @type {string}
-		 * @defaultvalue ""
-		 * @public
-		 * @since 1.0.0-rc.15
-		 */
-		accessibleNameRef: {
-			type: String,
-			defaultValue: "",
-		},
-
-		_iconPressed: {
-			type: Boolean,
-			noAttribute: true,
-		},
-
-		_filteredItems: {
-			type: Object,
-		},
-
-		_listWidth: {
-			type: Integer,
-			defaultValue: 0,
-			noAttribute: true,
-		},
-	},
-	managedSlots: true,
-	slots: /** @lends sap.ui.webc.main.ComboBox.prototype */ {
-		/**
-		 * Defines the component items.
-		 *
-		 * @type {sap.ui.webc.main.IComboBoxItem[]}
-		 * @slot items
-		 * @public
-		 */
-		"default": {
-			propertyName: "items",
-			type: HTMLElement,
-			invalidateOnChildChange: true,
-		},
-
-		/**
-		 * Defines the value state message that will be displayed as pop up under the component.
-		 * <br><br>
-		 *
-		 * <b>Note:</b> If not specified, a default text (in the respective language) will be displayed.
-		 * <br>
-		 * <b>Note:</b> The <code>valueStateMessage</code> would be displayed,
-		 * when the <code>ui5-combobox</code> is in <code>Information</code>, <code>Warning</code> or <code>Error</code> value state.
-		 * @type {HTMLElement[]}
-		 * @since 1.0.0-rc.9
-		 * @slot
-		 * @public
-		 */
-		valueStateMessage: {
-			type: HTMLElement,
-		},
-
-		/**
-		 * Defines the icon to be displayed in the input field.
-		 *
-		 * @type {sap.ui.webc.main.IIcon[]}
-		 * @slot
-		 * @public
-		 * @since 1.0.0-rc.9
-		 */
-		icon: {
-			type: HTMLElement,
-		},
-	},
-	events: /** @lends sap.ui.webc.main.ComboBox.prototype */ {
-		/**
-		 * Fired when the input operation has finished by pressing Enter, focusout or an item is selected.
-		 *
-		 * @event
-		 * @public
-		 */
-		change: {},
-
-		/**
-		 * Fired when typing in input.
-		 * <br><br>
-		 * <b>Note:</b> filterValue property is updated, input is changed.
-		 * @event
-		 * @public
-		 */
-		input: {},
-
-		/**
-		 * Fired when selection is changed by user interaction
-		 *
-		 * @event sap.ui.webc.main.ComboBox#selection-change
-		 * @param {HTMLElement} item item to be selected.
-		 * @public
-		 */
-		"selection-change": {
-			detail: {
-				item: { type: HTMLElement },
-			},
-		},
-	},
+const SKIP_ITEMS_SIZE = 10;
+
+type ValueStateAnnouncement = Record<ValueState, string | undefined>;
+type ValueStateTypeAnnouncement = Record<ValueState, string>;
+
+type ComboBoxListItem = StandardListItem & {
+	mappedItem: ComboBoxItem
+};
+
+enum ValueStateIconMapping {
+	Error = "error",
+	Warning = "alert",
+	Success = "sys-enter-2",
+	Information = "information",
+}
+
+type SelectionChangEventType = {
```

Comment:
As described in the ts dev article, to avoid type clashes, we add the component name in front of of the type:


type ComboBoxSelectionChangEventDetail


This is described in https://github.com/SAP/ui5-webcomponents/blob/main/docs/5-development/07-ts.md
look for "3. Create types for the Event Details."